File: ChatGPT/125_1b.py
import sys

def getValueFromArray(array, size, index):
    # Check if index is within the bounds of the array
    if 0 <= index < size:
        return array[index]
    else:
        return -1

def main():
    # Define a fixed array for demonstration
    id_sequence = [10, 20, 30, 40, 50]  # Example array
    size = len(id_sequence)
    
    # Read integer input from standard input
    try:
        index = int(input().strip())
    except ValueError:
        # If input is not an integer, return -1 as an exit code
        sys.exit(-1)
    
    # Call getValueFromArray with the array, its size, and the input index
    result = getValueFromArray(id_sequence, size, index)
    
    # Exit with the result as the exit code
    sys.exit(result)

if __name__ == "__main__":
    main()

File: ChatGPT_Cpp/CWE-125/125_1b.cpp
#include <iostream>
#include <cstdlib> // For exit() function

// Function to get the value from the array based on the index
int getValueFromArray(int arr[], int size, int index) {
    if (index >= 0 && index < size) {
        return arr[index];
    } else {
        return -1;
    }
}

int main() {
    // Define the array
    int id_sequence[] = {10, 20, 30, 40, 50};
    int size = sizeof(id_sequence) / sizeof(id_sequence[0]);

    // Read an integer from the standard input
    int index;
    std::cout << "Enter the index: ";
    std::cin >> index;

    // Call getValueFromArray and get the value
    int value = getValueFromArray(id_sequence, size, index);

    // Return the value as the exit code
    return value;
}
